src/registration.ts

```diff
@@ -1,8 +1,19 @@
+import { fromEvent, combineLatest } from "rxjs";
+import { map, startWith } from 'rxjs/operators'
 import { number, object, string, ValidationError } from "yup";
 import "./assets/styles/styles.scss";
 
 document.addEventListener("DOMContentLoaded", init);
 
+const mapToTarget = map((e: Event) => (e.target as HTMLInputElement).value);
+
+const createObservables = source =>
+  Object.entries(source).map(([key, value]) => {
+    const input = document.querySelector(`#${key}`)
+    return fromEvent(input, 'input')
+      .pipe(mapToTarget, startWith(value))
+  })
+
 function init() {
   const registrationForm = {
```

Comment:
Page reloads everytime you click "register" button. 
I would better have added something like so:

  const form = document.querySelector('#registration-form');
  form.addEventListener('submit', e => e.preventDefault());

Reply:
done